paper-api/src/main/java/org/bukkit/entity/Player.java

```diff
@@ -3477,6 +3477,20 @@ default void setNoTickViewDistance(int viewDistance) {
      */
     public void openSign(@NotNull Sign sign, @NotNull Side side);
 
+    /**
+     * Open a sign for editing by the player.
+     * <p>
+     * The sign must only be placed locally for the player, which can be done with {@link #sendBlockChange(Location, BlockData)} and {@link #sendBlockUpdate(Location, TileState)}.
+     * A side-effect of this is that normal events, like {@link org.bukkit.event.block.SignChangeEvent} will not be called (unless there is an actual sign in the world).
+     * </p>
```

Comment:
I think it should be noted that the fake sign needs to be within 9 blocks of the player.
(It's block interaction range + 4).

If it's too far away, the sign will immediately close.